### PR TITLE
Assorted usability fixes

### LINKED ISSUE #175: Make clipper_manager use version tag not latest versions of Docker containers
In clipper_manager, all the Docker containers use the `latest` tag when they should be used the `0.1` tag.

### LINKED ISSUE #176: Unloading old models
When a new model is deployed, can clipper unload previous versions of that model ?

We deploy at least 10 versions of a model a day. All the old docker containers hogs the machine. Right now we manually stop the old ones. Is it possible to have an API to undeploy?

### LINKED ISSUE #184: Remove uid from prediction queries until personalization is supported
Currently, users must supply a uid as part of the prediction request body but it is never used and must always be 0. We should remove the uid field entirely. The simplest way to do this is to just stop checking for `uid` when we parse the prediction request JSON and hardcode the uid as 0.

Replace [this line](https://github.com/ucbrise/clipper/blob/dd7b42185bc974b42cf104c2503f206477001bcb/src/frontends/src/query_frontend.hpp#L443) with

```cpp
long uid = 0;
```

and update the corresponding schema [here](https://github.com/ucbrise/clipper/blob/dd7b42185bc974b42cf104c2503f206477001bcb/src/frontends/src/query_frontend.hpp#L49-L54)

### PR DESCRIPTION
**Pull request by dcrankshaw**
This PR makes several backwards-compatible usability fixes to make Clipper less confusing. In particular:

+ Fixes #184: Removes uid from prediction query schema which was very confusing, because the uid must be set to 0 in the current version of Clipper. Now we just set it internally.
+ Fixes #176: Adds a method `remove_inactive_containers()` to clean up inactive containers (@rmdort)
+ Addresses several of the issues in #188 
  + imports the `Clipper` class in the `clipper_admin` module, so now users can import Clipper with `from clipper_admin import Clipper`
  + Makes `inspect_selection_policy` private so it doesn't show up in API documentation
  + Makes model labels optional and removes them from the examples
+ Adds a check to `clipper_admin` for python 3 and exits with an error message if Python 3 is detected.

### CODE REVIEW

File: clipper_admin/__init__.py
Comment (Corey-Zumar) [inline clipper_admin/__init__.py]: "Sorry, clipper_admin requires Python 2.x, but you are running Python 3.x" seems clearer

File: clipper_admin/clipper_manager.py
Comment (Corey-Zumar) [inline clipper_admin/clipper_manager.py:609]: Nit: Extra colon after `input`

File: clipper_admin/clipper_manager.py
Comment (Corey-Zumar) [inline clipper_admin/clipper_manager.py]: I'm assuming this is blocked by #187 and you'd like to merge this PR first. If not, we should use the branch associated with #187 as a base and add a test.

File: examples/tutorial/tutorial_part_two.ipynb
Comment (Corey-Zumar) [inline examples/tutorial/tutorial_part_two.ipynb:201]: Should the `cifar` and `sklearn` containers reference version `0.1` as opposed to `latest` as well?

File: clipper_admin/clipper_manager.py
Comment (Corey-Zumar) [inline clipper_admin/clipper_manager.py]: This fails in the case where no model containers are actively running.

```
>>> inst.remove_inactive_containers("m1")

Fatal error: local() encountered an error (return code 1) while executing 'docker inspect --format "{{ index .Config.Labels \"ai.clipper.model_container.model_version\"}}" '

================================ Standard error ================================

"docker inspect" requires at least 1 argument(s).
See 'docker inspect --help'.

Usage:  docker inspect [OPTIONS] NAME|ID [NAME|ID...]

Return low-level information on Docker objects

================================================================================

Aborting.
```

File: clipper_admin/clipper_manager.py
Comment (Corey-Zumar) [inline clipper_admin/clipper_manager.py]: `tar_loc` is now undefined yet is being referenced. Why were these lines removed?

File: examples/basic_query/example_client.py
Comment (Corey-Zumar) [inline examples/basic_query/example_client.py:25]: why don't we check for docker here?

**Review by Corey-Zumar — CHANGES_REQUESTED** [review]

**Review by jegonzal — COMMENTED** [review]
@Corey-Zumar did an excellent job reviewing.  I would approve pending his comments.

File: clipper_admin/__init__.py
Comment (dcrankshaw) [reply clipper_admin/__init__.py]: Done

File: clipper_admin/clipper_manager.py
Comment (dcrankshaw) [reply clipper_admin/clipper_manager.py:609]: That's an intentional pydoc markup symbol (https://docs.python.org/devguide/documenting.html#source-code)

File: clipper_admin/clipper_manager.py
Comment (dcrankshaw) [reply clipper_admin/clipper_manager.py]: Yeah I want to add a test for this. Let's get #187 merged before this. Leaving the TODO in for now.

File: clipper_admin/clipper_manager.py
Comment (dcrankshaw) [reply clipper_admin/clipper_manager.py]: Oh weird. They were not supposed to be. Adding them back.

File: examples/basic_query/example_client.py
Comment (dcrankshaw) [reply examples/basic_query/example_client.py:25]: Because this example doesn't start Clipper, it assumes there is already a Clipper instance running and model connected. We don't check for Docker so that the example works even if you aren't using Docker to run Clipper.

File: examples/tutorial/tutorial_part_two.ipynb
Comment (dcrankshaw) [reply examples/tutorial/tutorial_part_two.ipynb:201]: I don't think it matters for this tutorial.

File: clipper_admin/clipper_manager.py
Comment (dcrankshaw) [reply clipper_admin/clipper_manager.py]: Good catch. Fixed.

File: clipper_admin/clipper_manager.py
Comment (Corey-Zumar) [reply clipper_admin/clipper_manager.py:609]: Got it, thanks for clarifying!

File: examples/tutorial/tutorial_part_two.ipynb
Comment (Corey-Zumar) [reply examples/tutorial/tutorial_part_two.ipynb:201]: Agreed

File: examples/basic_query/example_client.py
Comment (Corey-Zumar) [reply examples/basic_query/example_client.py:25]: Got it! The key part is the docker-independent functionality.

**Review by Corey-Zumar — APPROVED** [review]
LGTM. Let me know if I should hold off until #187 is merged. If not, we should create an issue reminding us to write a test for `remove_inactive_containers`.

**Comment by dcrankshaw** [comment]
@Corey-Zumar this went through a big round of changes, so you'd better review again

File: bin/run_unittests.sh
Comment (Corey-Zumar) [inline bin/run_unittests.sh]: Now that we have multiple integration tests, let's rename `light_load_all_functionality` to something more descriptive.

File: clipper_admin/clipper_manager.py
Comment (feynmanliang) [inline clipper_admin/clipper_manager.py]: What's the purpose of having all models get a "default" label? Can this be defaulted to `[]`?

File: clipper_admin/clipper_manager.py
Comment (feynmanliang) [inline clipper_admin/clipper_manager.py:1089]: What is the reasoning for dropping into shell commands over using libraries? e.g. https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.list

File: src/frontends/src/query_frontend_tests.cpp
Comment (Corey-Zumar) [inline src/frontends/src/query_frontend_tests.cpp:102]: Very nit: Typo in "punctuation".

**Review by Corey-Zumar — CHANGES_REQUESTED** [review]
Just requesting that we rename an integration test file + a nit comment. Looks good in general

File: integration-tests/clipper_manager_tests.py
Comment (feynmanliang) [inline integration-tests/clipper_manager_tests.py]: AFAICT `clipper_manager` is only used to access the `Clipper` class so we could directly import `Clipper` here

File: integration-tests/light_load_all_functionality.py
Comment (feynmanliang) [inline integration-tests/light_load_all_functionality.py]: Ditto about importing `Clipper` directly

File: clipper_admin/clipper_manager.py
Comment (dcrankshaw) [reply clipper_admin/clipper_manager.py:1089]: Simplicity and wanting to use the same commands locally and remotely. As we refactor clipper_manager.py we should definitely look into the Python SDK.

File: bin/run_unittests.sh
Comment (dcrankshaw) [reply bin/run_unittests.sh]: Done

File: src/frontends/src/query_frontend_tests.cpp
Comment (dcrankshaw) [reply src/frontends/src/query_frontend_tests.cpp:102]: This isn't a real word anyway. Leaving as is.

File: integration-tests/clipper_manager_tests.py
Comment (dcrankshaw) [reply integration-tests/clipper_manager_tests.py]: Ahh yeah good catch. This is a holdover from before the package structure existed.

**Comment by dcrankshaw** [comment]
@Corey-Zumar @feynmanliang addressed your comments. @Corey-Zumar please merge once the tests pass.

**Review by Corey-Zumar — APPROVED** [review]
LGTM

**Comment by dcrankshaw** [comment]
jenkins test this please

**Pull request merged**
